ntoskrnl/io/pnpmgr/pnpirp.c

```diff
@@ -306,3 +306,61 @@ PiIrpQueryPnPDeviceState(
 
     return status;
 }
+
+// IRP_MN_QUERY_CAPABILITIES (0x09)
+NTSTATUS
+PiIrpQueryPnPDeviceCapabilities(
```

Comment:
It is a good practice that newly added functions in the kernel have to be documented using Doxygen format, so that people can understand more or less how the newly implemented routines in question work or their purpose.

https://reactos.org/wiki/Documentation_Guidelines

See an example -> https://github.com/reactos/reactos/blob/master/ntoskrnl/se/tokenadj.c#L17